### PR TITLE
update icewind/smb to 3.1.0

### PR DESCRIPTION
**Pull request by icewind1991**
relevant fixes are

- improved multibyte path support
- improved timezone detection
- fix incorrect filesize for files with multiple streams 

I'll probably manually backport the timezone fixes

### CODE REVIEW

File: apps/files_external/3rdparty/composer/autoload_classmap.php [hunk 1]
```diff
@@ -49,8 +52,19 @@
...
+    'Icewind\\SMB\\Test\\AbstractShareTest' => $vendorDir . '/icewind/smb/tests/AbstractShareTest.php',
+    'Icewind\\SMB\\Test\\NativeShareTest' => $vendorDir . '/icewind/smb/tests/NativeShareTest.php',
+    'Icewind\\SMB\\Test\\NativeStreamTest' => $vendorDir . '/icewind/smb/tests/NativeStreamTest.php',
+    'Icewind\\SMB\\Test\\NotifyHandlerTest' => $vendorDir . '/icewind/smb/tests/NotifyHandlerTest.php',
+    'Icewind\\SMB\\Test\\ParserTest' => $vendorDir . '/icewind/smb/tests/ParserTest.php',
+    'Icewind\\SMB\\Test\\ServerFactoryTest' => $vendorDir . '/icewind/smb/tests/ServerFactoryTest.php',
+    'Icewind\\SMB\\Test\\ServerTest' => $vendorDir . '/icewind/smb/tests/ServerTest.php',
+    'Icewind\\SMB\\Test\\ShareTest' => $vendorDir . '/icewind/smb/tests/ShareTest.php',
+    'Icewind\\SMB\\Test\\TestCase' => $vendorDir . '/icewind/smb/tests/TestCase.php',
+    'Icewind\\SMB\\Test\\TimeZoneProviderTest' => $vendorDir . '/icewind/smb/tests/TimeZoneProviderTest.php',
```
Comment (MorrisJobke) [inline apps/files_external/3rdparty/composer/autoload_classmap.php]: Could the tests be removed?

File: apps/files_external/3rdparty/icewind/smb/.editorconfig [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+tab_width=4
+
+[{*.module,*.hphp,*.phtml,*.php5,*.php4,*.php,*.inc}]
+indent_style=tab
+tab_width=4
+
+[{*.yml,*.yaml}]
+indent_style=space
+indent_size=2
+
```
Comment (MorrisJobke) [inline apps/files_external/3rdparty/icewind/smb/.editorconfig]: Also this file can be removed from git.

Comment (icewind1991) [reply apps/files_external/3rdparty/icewind/smb/.editorconfig]: done

File: apps/files_external/3rdparty/composer/autoload_classmap.php (hunk 1 above)
Comment (icewind1991) [reply apps/files_external/3rdparty/composer/autoload_classmap.php]: done

**Review by MorrisJobke — APPROVED** [review]

**Review by rullzer — APPROVED** [review]

**Pull request merged**